src/lib/hooks/swap/useConvenienceFee.ts

```diff
@@ -0,0 +1,43 @@
+import { Percent } from '@uniswap/sdk-core'
+import useActiveWeb3React from 'hooks/useActiveWeb3React'
+import { useAtom } from 'jotai'
+import { feeOptionsAtom } from 'lib/state/swap'
+import { useCallback, useEffect } from 'react'
+
+interface FeeOptionsArgs {
+  convenienceFee?: number
+  convenienceFeeRecipient?: string | string | { [chainId: number]: string }
+}
+
+export default function useConvenienceFee({ convenienceFee, convenienceFeeRecipient }: FeeOptionsArgs) {
+  const { chainId } = useActiveWeb3React()
+  const [feeOptions, updateFeeOptions] = useAtom(feeOptionsAtom)
+
+  const setToDefaults = useCallback(() => {
+    updateFeeOptions({
+      fee: new Percent(0),
+      recipient: '',
+    })
+  }, [updateFeeOptions])
+
+  useEffect(() => {
+    if (convenienceFee && convenienceFeeRecipient) {
```

Comment:
You should derive the convenience fee as a percent once, here, and it should follow this pattern:

https://github.com/Uniswap/interface/blob/main/src/lib/components/Swap/Settings/MaxSlippageSelect.tsx#L112-L114

I think that you are dividing the intended amount by 100 here.

Reply:
The integrator is passing an integer value for bips, so if we use that as the numerator we need to divide by 10,000 for a valid %. i.e., 100/10_000 = 1%